### PR TITLE
Cmr 4703

### PR DESCRIPTION
**Pull request by littleidiot40**

### CODE REVIEW

**Comment by oubiwann** [comment]
Huh, looks like this needs a rebase ...

File: collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj [hunk 1]
```diff
@@ -107,15 +108,21 @@
...
-  [context collection]
+  [context collection concept-id]
   (let [umm-json (umm-json/umm->json
                   (vm/migrate-umm context
                                   :collection
                                   umm-version/current-collection-version
                                   (context->preview-gem-umm-version context)
                                   collection))]
+    (defconfig search-edsc-url
+         "URL of the Earthdata Search application"
```
Comment (oubiwann) [inline collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj]: What are your editor settings for tab/space indent? The two lines after the `defconfig` need to be indented just two spaces (there should be no tabs in our Clojure files; can't tell what's used by looking at the diff on Github).

File: collection-renderer-lib/test/cmr/collection_renderer/test/services/collection_renderer.clj [hunk 2]
```diff
@@ -22,20 +22,27 @@
...
   (for-all [umm-record (gen/no-shrink umm-gen/umm-c-generator)]
-    (let [^String html (cr/render-collection (renderer-context) umm-record)]
+    (let [^String html (cr/render-collection (renderer-context) umm-record "C-1234")]
       (and html (.contains html (:EntryTitle umm-record))))))
 
 ;; Verify that the title of the html is the entry title
 (deftest render-title-as-entry-title
   (let [coll expected-conversion/example-collection-record
-         ^String html (cr/render-collection (renderer-context) coll)]
+         ^String html (cr/render-collection (renderer-context) coll "C-1234")]
```
Comment (oubiwann) [inline collection-renderer-lib/test/cmr/collection_renderer/test/services/collection_renderer.clj]: looks like this line needs to be un-indented by one space ...

File: collection-renderer-lib/test/cmr/collection_renderer/test/services/collection_renderer.clj [hunk 3]
```diff
@@ -22,20 +22,27 @@
...
     (is html "Expected a title containing the entry title")
-    (is (.contains html (str "<title>" (:EntryTitle coll) "</title>")))))
\ No newline at end of file
+    (is (.contains html (str "<title>" (:EntryTitle coll) "</title>")))))
+
+;; Verify that the correct EDSC url is rendered
+(deftest render-edsc-link
+  (let [coll expected-conversion/example-collection-record
+         ^String html (cr/render-collection (renderer-context) coll "C-1234")]
```
Comment (oubiwann) [inline collection-renderer-lib/test/cmr/collection_renderer/test/services/collection_renderer.clj]: Same: unindent by one.

**Comment by oubiwann** [comment]
Rebase looks good!

**Comment by littleidiot40** [comment]
Indent changes made

File: collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj [hunk 4]
```diff
@@ -107,15 +108,21 @@
...
   "Renders a UMM-C collection record and returns the HTML as a string."
-  [context collection]
+  [context collection concept-id]
   (let [umm-json (umm-json/umm->json
                   (vm/migrate-umm context
                                   :collection
                                   umm-version/current-collection-version
                                   (context->preview-gem-umm-version context)
                                   collection))]
+    (defconfig search-edsc-url
```
Comment (chris-durbin) [inline collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj]: The `defconfig` should be pulled out of the render-collection function (define this above render-collection on line 108).

File: system-int-test/test/cmr/system_int_test/search/collection_concept_retrieval_test.clj [hunk 5]
```diff
@@ -164,6 +164,7 @@
                                                 :url-extension "html"})
             _ (is (= 200 (:status response)))]
         (is (search/mime-type-matches-response? response mt/html))
+        (is (= "C1200000016-PROV1" (:concept-id coll1) ))
```
Comment (chris-durbin) [inline system-int-test/test/cmr/system_int_test/search/collection_concept_retrieval_test.clj]: We should update this test to not rely on the autogenerated concept-ids (we can do that outside of this PR, and I see line 139 does the same thing). The way to fix this is to specify the concept-id as part of the ingest request on lines 118 and 119.

Comment (chris-durbin) [reply system-int-test/test/cmr/system_int_test/search/collection_concept_retrieval_test.clj]: Can you actually change this check to verify the concept-id is in the HTML response?
`(is (.contains ^String (:body response) (:concept-id coll1)))))`

File: collection-renderer-lib/test/cmr/collection_renderer/test/services/collection_renderer.clj [hunk 6]
```diff
@@ -22,20 +22,27 @@
 
 (deftest render-default-collection
   (let [coll expected-conversion/example-collection-record
-        ^String html (cr/render-collection (renderer-context) coll)]
+        ^String html (cr/render-collection (renderer-context) coll "C-1234")]
```
Comment (chris-durbin) [inline collection-renderer-lib/test/cmr/collection_renderer/test/services/collection_renderer.clj]: Nit: concept ID format would be something like C1234-PROV1.

**Pull request merged**